examples/asr/speech_to_text_bpe.py

```diff
@@ -0,0 +1,112 @@
+# Copyright (c) 2020, NVIDIA CORPORATION.  All rights reserved.
```

Comment:
I have not looked into this PR in depth, but if we require both a speech_to_text_bpe.py and a speech_to_text.py, we need to stop and reconsider our design of these two scripts. There should just be one speech_to_text.py.

Reply:
Also speech_to_text.py is bad name unless we plan on putting ALL future ASR models under one script, we need to reconsider our naming of this script.

Reply:
This is a good point. I grappled with a design decision - should I add flags in config to turn on and off BPE, to turn on and off other types than CtC in the future etc, or to have separate scripts for all. I am fine with merging it, but it adds significant switching complexity to the train script for all the current and future types of models

Reply:
What flags are you considering switching between?
As we discussed in the DictConfig PR, it would make sense for users to simply pass in a base configuration object which points to the hparams of a model.

Reply:
I was considering a flag in the data preprocessing config - pass an addition value to constructor "tokenizer_pat", resolve at runtime to check if none or not, and switch between the two different EncDecConvModel or EncDecConvModel or based on that.

Reply:
Doesn't hydra or omegaconf allow us to specify the class in the yaml itself? Is it possible to add `cls` to model and smartly call whichever one is needed?

Reply:
Yes, but then the speech_to_text.py will need to import both the ASR only speech to text + the ASR + NLP collection speech to text bpe layers, which means by default our asr scripts will need to install as nemo-toolkit[asr,nlp]. I think its preferable to keep them separate for now.

Reply:
>I have not looked into this PR in depth, but if we require both a speech_to_text_bpe.py and a speech_to_text.py, we need to stop and reconsider our design of these two scripts. There should just be one speech_to_text.py.

- I am not sure about this. I think char-based vocab vs BPE-based vocab is significant enough conceptual difference to warrant own script. So this, imho, is OK. We certainly should no have jasper/quartznet though because there the difference is really tiny

Reply:
especially because this now needs tokenizer